integration/test/performance_tracker/transmitter/transmitter.go

```diff
@@ -0,0 +1,237 @@
+	Data []float64
+}
+
+type collectorData []struct{ // this is the struct data collector passes in
+	Id string	`json:"Id"`
+	Label string	`json:Label`	
+	Messages string	`json:Messages`
+	StatusCode string	`json:StatusCode`
+	Timestamps []string	`json:Timestamps`
+	Values []float64	`json:Values`
+}
+
+
+/*
+InitializeTransmitterAPI
+Desc: Initializes the transmitter class
+Side effects: Can create a dynamodb table
+*/
+func InitializeTransmitterAPI(DataBaseName string) * TransmitterAPI{
+	transmitter := new(TransmitterAPI)
+	//setup aws session
+	session := session.Must(session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	}))
+	transmitter.DynamoDbClient =  dynamodb.New(session)
+	transmitter.DataBaseName = DataBaseName //fmt.Sprintf("%d",int(time.Now().UnixNano()))
+	// check if the dynamo table exist if not create it
+	tableExist, err:= transmitter.TableExist()
+	if err !=nil{
+		return nil
+	}
+	if !tableExist{
+		fmt.Println("Table doesn't exist")
+		err := transmitter.CreateTable()
+		if err != nil{
+			fmt.Println("Couldn't create table")
+			return nil
+		}
+	}
+	fmt.Println("API ready")
+	return transmitter
+
+}
+/*
+CreateTable()
+Desc: Will create a DynamoDB Table with given param. and config
+Params:
+Side Effects: Creates a dynamoDB table
+*/
+func (transmitter * TransmitterAPI) CreateTable() error{
+	input := &dynamodb.CreateTableInput{
+		AttributeDefinitions: []*dynamodb.AttributeDefinition{
+			{
+				AttributeName: aws.String("Hash"),
+				AttributeType: aws.String("S"),
+			},
+		},
+		KeySchema: []*dynamodb.KeySchemaElement{
+			{
+				AttributeName: aws.String("Hash"),
+				KeyType:       aws.String("HASH"),
+			},
+		},
+		ProvisionedThroughput: &dynamodb.ProvisionedThroughput{
+			ReadCapacityUnits:  aws.Int64(10),
+			WriteCapacityUnits: aws.Int64(10),
+		},
+		TableName: aws.String(transmitter.DataBaseName),
+	} // this is the config for the new table
+	
+	_, err := transmitter.DynamoDbClient.CreateTable(input)
+	if err != nil {
+		fmt.Printf("Got error calling CreateTable: %s", err)
+		return err
+	}
+	
+	fmt.Println("Created the table", transmitter.DataBaseName)
+	return nil
+}
+/*
+AddItem() 
+Desc: Takes in a packet and
+will convert to dynamodb format  and upload to dynamodb table.
+Param: 
+	packet * map[string]interface{}:  is a map with data collection data
+Side effects: 
+	Adds an item to dynamodb table
+*/
+func (transmitter * TransmitterAPI) AddItem(packet map[string]interface{})(string,error){
+	// fmt.Printf("Packet: %+v \n",packet)
+	// metrics, _ := dynamodbattribute.MarshalMap(packet.Metrics)
+	// fmt.Printf("Metric: %+v \n OG:%+v \n",metrics, packet.Metrics)
+	item, err := dynamodbattribute.MarshalMap(packet)
+	if err != nil {
+		panic(err)
+	}
+	DBitem := &dynamodb.PutItemInput{
+		Item:      item,
+		TableName: aws.String(transmitter.DataBaseName),
+	}
+	// fmt.Println(DBitem)
+	// return err
+	_, err = transmitter.DynamoDbClient.PutItem(DBitem)
```

Comment:
Luckily, I think we're fine here because if we reuse a key, PutItem replaces the whole item, which is desirable for our usage of DDB.